src/applications/mhv-secure-messaging/components/ComposeForm/ComposeForm.jsx

```diff
@@ -513,38 +529,78 @@ const ComposeForm = props => {
         category === null &&
         attachments.length === 0;
 
-      const isSavedEdits = () =>
+      const isEditedSaved = () =>
         messageBody === draft?.body &&
         Number(selectedRecipientId) === draft?.recipientId &&
         category === draft?.category &&
         subject === draft?.subject;
 
-      const isEditPopulatedForm = () =>
+      const isEditedForm = () =>
         (messageBody !== draft?.body ||
           selectedRecipientId !== draft?.recipientId ||
           category !== draft?.category ||
           subject !== draft?.subject) &&
         !isBlankForm() &&
-        !isSavedEdits();
+        !isEditedSaved();
 
-      const unsavedDraft = isEditPopulatedForm() && !deleteButtonClicked;
-      if (!isEditPopulatedForm() || !isSavedEdits()) {
-        setSavedDraft(false);
-      }
+      const isFormFilled = () =>
+        messageBody !== '' &&
+        subject !== '' &&
+        selectedRecipientId !== null &&
+        category !== null;
 
       let error = null;
-      if (isBlankForm() || savedDraft) {
+      const unsavedFilledDraft =
+        isFormFilled() && !isEditedSaved() && !savedDraft;
+
+      const partiallySavedDraftWithSignRequired =
+        !draft &&
+        unsavedFilledDraft &&
+        !attachments.length &&
+        isSignatureRequired;
+
+      const partiallySavedDraft =
+        (!isFormFilled() && (!isBlankForm() || attachments.length > 0)) ||
+        partiallySavedDraftWithSignRequired;
+
+      const savedDraftWithEdits =
+        (savedDraft && !isEditedSaved() && isEditedForm()) ||
+        (!!draft && unsavedFilledDraft);
+
+      const savedDraftWithNoEdits =
+        (savedDraft && !isEditedForm()) || (!!draft && !isEditedForm());
+
```

Comment:
worth noting that updated variable names make code more comprehensive. Thanks a lot for doing this!